design-documents/graph-ql/coverage/add-items-to-cart-single-mutation.md

```diff
@@ -0,0 +1,205 @@
+# Single mutation for adding products to cart
+In order to simplify the flow of adding different type of products to wishlist and cart it is proposed to use single mutation for all product types.
+
+*Proposed schema for adding products to cart* - [AddProductsToCart](AddProductsToCart.graphqls)
+
+*Proposed schema for adding products to wish list* - [AddProductsToWishlist](AddProductsToWishlist.graphqls)
+
+## Definition
+
+- Bundle dynamic, grouped and configurable product are just templates which help merchant to select a set of simple products.
+- Product may have selected options and entered options (buyer input).
+- Options are sets of predefined values that a buyer can add to cart/wishlist in addition to a product.
+- The difference between wishlist and cart payloads is that all fields in wishlist are optional.
+
+### Examples
+
+#### Add simple product to cart
+```
+[
+  {
+    "sku": "Cup",
+    "qty": 1,
+    "selected_options" : [
+      base64_encode('configurable/24/42')
```

Comment:
We may introduce shortened names for 'customizable-option', 'configurable-option' etc. E.g. 'cst', 'cnf', 'bnd', 'dwn'